File: homework4_1/src/main/java/by/itacademy/homework4_1/ChangeItemActivity.java
package by.itacademy.homework4_1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class ChangeItemActivity extends AppCompatActivity {
    private EditText name;
    private EditText info;
    private int position;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_change_item);

        final Intent intent = getIntent();
        if (intent != null) {
            position = intent.getIntExtra("position", 0);
        }
        name = findViewById(R.id.name);
        info = findViewById(R.id.info);
        Contact contact = Publisher.getInstance().getContactList().get(position);
        name.setText(contact.getName());
        String data = contact.getData();
        if (data != null) {
            info.setText(contact.getData());
        }
        findViewById(R.id.back).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Contact newContact = new Contact();
                newContact.setName(name.getText().toString());
                newContact.setData(info.getText().toString());
                Publisher.getInstance().setItem(position, newContact);
                finish();
            }
        });
        findViewById(R.id.remove).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Publisher.getInstance().removeItem(position);
                finish();
            }
        });
    }
}

File: app/contacts/src/main/java/by/itacademy/contacts/view/ChangeItemActivity.kt
package by.itacademy.contacts.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import by.itacademy.contacts.data.Contact
import by.itacademy.contacts.presentation.DBOperations
import by.itacademy.contacts.presentation.DBOperationsImpl
import by.itacademy.contacts.databinding.ActivityChangeItemBinding


class ChangeItemActivity : AppCompatActivity() {
    private lateinit var binding: ActivityChangeItemBinding
    private val dbOperations: DBOperations = DBOperationsImpl()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityChangeItemBinding.inflate(layoutInflater)
        val position = intent?.getIntExtra("position", 0) ?: -1
        val contact = dbOperations.getUsersFromDB(applicationContext)[position]
        with(binding) {
            name.setText(contact.name)
            info.setText(contact.data)
        }
        changeContact(contact, position)
        removeContact(position)
        setContentView(binding.root)
    }

    private fun changeContact(contact: Contact, position: Int) {
        binding.back.setOnClickListener {
            val newContact = Contact().apply {
                name = binding.name.text.toString()
                data = binding.info.text.toString()
                isPhone = contact.isPhone
            }
            dbOperations.changeContact(applicationContext, newContact, position)
            finish()
        }
    }

    private fun removeContact(position: Int) {
        binding.remove.setOnClickListener {
            dbOperations.removeContact(applicationContext, position)
            finish()
        }
    }
}